packages/medusa/src/api/middlewares/transform-query.ts

```diff
@@ -78,12 +92,22 @@ export function transformStoreQuery<
       )
       req.validatedQuery = validated
       req.filterableFields = getFilterableFields(validated)
-      req.allowedProperties = getStoreAllowedProperties(
-        validated,
-        req.includes ?? {},
-        queryConfig
-      )
       attachListOrRetrieveConfig<TEntity>(req, queryConfig)
+      /**
+       * TODO: shouldn't this correspond to returnable fields instead of allowed fields? also it is used by the cleanResponseData util and is supposed to
+       * only return the fields that are allowed to be queried either from the default of from what the user requested.
+       * The validation occurred above and the fields are already validated and should be used to filter the response.
+       */
+      const queryConfigRes = req.retrieveConfig ?? req.listConfig
```

Comment:
Should we exctract this logic into a function? I see it occuring twice

Reply:
I ve done a larger cleanup as now both middlewares are the same, only the API change